assets/app/scripts/directives/fromFile.js

```diff
@@ -0,0 +1,318 @@
+              message: "Resource is missing metadata field."
+            };
+            return true;
+          }
+          if (item.metadata.namespace && item.metadata.namespace !== $scope.projectName) {
+            $scope.error = {
+              message: item.kind + " " + item.metadata.name + " can't be created in project " + item.metadata.namespace + ". Can't create resource in different projects."
+            };
+            return true;
+          }
+          return false;
+
+        }
+
+        function openTemplateProcessModal() {
+          var modalInstance = $uibModal.open({
+            animation: true,
+            templateUrl: 'views/modals/process-template.html',
+            controller: 'ProcessTemplateModalController',
+            scope: $scope
+          });
+          modalInstance.result.then(function() {
+            if ($scope.templateOptions.process) {
+              CachedTemplateService.setTemplate($scope.resourceList[0]);
+            }
+            if ($scope.templateOptions.add) {
+              createAndUpdate();
+            } else {
+              redirect();
+            }
+          });      
+        }
+
+        function confirmReplace() {
+          var modalInstance = $uibModal.open({
+            animation: true,
+            templateUrl: 'views/modals/confirm-replace.html',
+            controller: 'ConfirmReplaceModalController',
+            scope: $scope
+          });
+          modalInstance.result.then(function() {
+            createAndUpdate();
+          });
+        }
+
+        function createAndUpdate() {
+          var createUpdatePromises = [];
+          if ($scope.updateResources.length > 0) {
+            createUpdatePromises.push(updateResourceList());
+          }
+          if ($scope.createResources.length > 0) {
+            createUpdatePromises.push(createResourceList());
+          }
+          $q.all(createUpdatePromises).then(redirect);
+        }
+
+        function redirect() {
+          var path;
+          if ($scope.resourceKind === "Template" && $scope.templateOptions.process) {
+            path = Navigate.fromTemplateURL($scope.projectName, $scope.resourceName);
+          } else {
+            // Navigate.toProjectOverviewURL($scope.projectName);
+            path = Navigate.projectOverviewURL($scope.projectName);
+          }
+          $location.url(path);
+        }
+
+        function checkResource(item) {
+          // Check if the resource already exists. If it does, replace it spec with the new one.
+          return DataService.get(APIService.kindToResource(item.kind), item.metadata.name, $scope.context, {errorNotification: false}).then(
+            // resource does exist
+            function(resource) {
+              resource.spec = item.spec;
+              $scope.updateResources.push(resource);
+            },
+            // resource doesn't exist
+            function() {
+              $scope.createResources.push(item);
```

Comment:
We should probably check that the error has status code 404 and not some other problem.